cpp/src/link_prediction/similarity_impl.cuh

```diff
@@ -368,187 +368,196 @@ all_pairs_similarity(raft::handle_t const& handle,
       sum_two_hop_degrees,
       MAX_PAIRS_PER_BATCH);
 
```

Comment:
In the lines above,

```
    top_v1.reserve(*topk, handle.get_stream());
    top_v2.reserve(*topk, handle.get_stream());
    top_score.reserve(*topk, handle.get_stream());
```
Shouldn't `reserve` here be `resize`?

Reply:
```
    raft::update_host(&sum_two_hop_degrees,
                      two_hop_degree_offsets.data() + two_hop_degree_offsets.size() - 1,
                      1,
                      handle.get_stream());
```
We are missing `handle.sync_stream()` after this to ensure that `sum_two_hop_degrees` is ready to use in the following `compute_offset_aligned_element_chunks`.

Reply:
I think this is the big thing that I was tripping over.  Added the sync (along with fixing the problem you note below) and the hang appears to be resolved.